builder/dockerfile/instructions/parse.go

```diff
@@ -228,9 +228,31 @@ func parseLabel(req parseRequest) (*LabelCommand, error) {
 		return nil, err
 	}
 
+	kvpMap := make(map[string]string)
```

Comment:
What is the reason for the lines 231-251 here? Why wouldn't just setting `expand` to true work?

Reply:
The reason for the lines 231-251 is to make sure the env variable LABEL like this "LABEL foo="$PATH" don't expand, so we can get the expected "$PATH" while not something like "/usr/bin:/usr/local/bin" because `expand=true` will trigger the expand handling method if no single quoted way by lines 231-251, which is different from the `--label` option where the `expand=false` meaning the env variable will not be expanded.

Reply:
But the variables are expanded to `LABEL` values with the current implementation. If you think this should be changed it doesn't belong in this PR.

Reply:
>But the variables are expanded to LABEL values with the current implementation

Don't quite understand here. If the `current implementation` is the PR's, then the env variables will not be expanded, this is the same as the original code before this PR. Correct me if I am wrong here.

>If you think this should be changed it doesn't belong in this PR.

If we don't have this change in this PR, then we will have 2 label related test cases failure as we saw before.
But if you have some more reasonable solution, I am happy to implement that.

Reply:
```
/tmp # cat Dockerfile
from busybox
label foo="$PATH"
/tmp # docker inspect -f '{{.Config.Labels}}' $(docker build -q .)
map[foo:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin]
/tmp # docker version -f '{{ .Server.Version}}'
18.03.1-ce
```

Reply:
Ah, that's because the client side will expand the env variable automatically  and `POST` the expanded `LABEL` value to the server, if we apply this PR, the result is the same as `18.03.1-ce`

Reply:
No, we definitely do not evaluate any variables inside a Dockerfile in client side.

Reply:
What I said is the client  side will expand the env variable of `LABEL` command then `POST` it to the server, look like this: 
```
Calling POST /v1.35/build?buildargs=%7B%7D&cachefrom=%5B%5D&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile&labels=%7Buser%7Dbin%7D...
```

Reply:
`labels=` in the POST query are the extra labels from `--label` flag(not expanded by client as well, could be by the shell when using cli but that is irrelevant) not the values from `LABEL` command. Even with this PR these never call this `parseLabel` function that is being discussed here.

Reply:
Hmm, I just checked the debug message of the daemon, seems the `--label foo="$PATH"` will expand to `labels=%7B%22bbc%22%3A%22%2Fgo%2Fbin%3A%2Fusr%2Flocal%2Fsbin...` when `POST`, the `LABEL` inside the Dockerfile will not. Sorry for the misleading info 😄

Reply:
But the `LABEL foo="$PATH"` inside the Dockerfile has the same behavior before and after this PR

Reply:
So @tonistiigi , now we can focus on the `parseLabel` function, if we just set `expand=true`, we will fail the CI as you know 😭

Reply:
What test will fail and why?